### PR TITLE
Create NotificationAvatarsComponent

### PR DESCRIPTION
**Pull request by dmarcoux**
The partials and the presenter made this difficult to cover with unit tests.

Part of #11558

Based on what Saray did in #12018.

### CODE REVIEW

**Comment by dmarcoux** [comment]
[Turn NotificationPresenter into a view component (M)](https://trello.com/c/U8c9ghnP/1836-turn-notificationpresenter-into-a-view-component-m)

**Review by saraycp — APPROVED** [review]

File: src/api/app/components/notification_avatars_component.html.haml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        +#{number_of_hidden_users}
+  - avatars_to_display.each do |avatar_object|
+    %li.list-inline-item
+      - case avatar_object.class.name
+      - when 'User', 'Group'
+        = helpers.image_tag_for(avatar_object, size: 23, custom_class: 'rounded-circle bg-light border border-gray-400')
+      - when 'Package'
+        %span.fa.fa-archive.text-warning.rounded-circle.bg-light.border.border-gray-400.simulated-avatar{ title: package_title(avatar_object) }
+      - when 'Project'
+        %span.fa.fa-cubes.text-secondary.rounded-circle.bg-light.border.border-gray-400.simulated-avatar{ title: "Project #{avatar_object}" }
```
Comment (saraycp) [inline src/api/app/components/notification_avatars_component.html.haml:14]: :right_anger_bubble: Just for little more consistency, I'd also call a method here.
```suggestion
        %span.fa.fa-cubes.text-secondary.rounded-circle.bg-light.border.border-gray-400.simulated-avatar{ title: project_title(avatar_object) }
```

Comment (dmarcoux) [reply src/api/app/components/notification_avatars_component.html.haml]: Suggestion applied.

**Review by saraycp — APPROVED** [review]

**Comment by saraycp** [comment]
Flickering test, re-running...

**Comment by dmarcoux** [comment]
I had to rebase on master to fix a conflict after another PR was merged.

**Pull request merged**